transaction/core/src/ring_signature/rct_bulletproofs.rs

```diff
@@ -347,46 +560,70 @@ fn sign_with_balance_check<CSPRNG: RngCore + CryptoRng>(
         }
     }
 
-    let pseudo_output_commitments: Vec<CompressedCommitment> = commitments
-        .iter()
-        .take(num_inputs)
-        .map(CompressedCommitment::from)
+    // The actual pseudo output commitments use the blindings from
+    // `pseudo_output_blinding` and not the original true input.
+    let pseudo_output_commitments: Vec<CompressedCommitment> = pseudo_output_commitments
```

Comment:
Repurposing the `pseudo_output_commitments` here seems a bit janky, any reason not to `CompressedCommitment::from` at the block that generates the original `pseudo_output_commitments` from `input_secrets`/`pseudo_output_blindings`?

Reply:
we need to add them at line 550, and they can't be compressed to add them

we could change the name of this variable if you think that's better

Reply:
ah I see. I don't feel strongly about it.